app/javascript/leaflet/viewers/tilejson.js

```diff
@@ -0,0 +1,31 @@
+import { latLng, latLngBounds, tileLayer } from "leaflet";
+import LeafletViewerWms from "./wms.js";
+
+export default class LeafletViewerTileJson extends LeafletViewerWms {
+  addPreviewLayer() {
+    fetch(this.data.url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Unable to fetch tile.json document");
+        }
+        return response.json();
+      })
+      .then((tileJson) => {
+        const bounds = this.getBounds(tileJson),
+          options = bounds ? { bounds } : {},
+          url = tileJson.tiles[0],
+          tileJsonLayer = tileLayer(url, options);
+        this.overlay.addLayer(tileJsonLayer);
+      })
+      .catch((error) => console.debug(error));
+  }
+
+  getBounds(doc) {
+    const bounds = doc.bounds;
+    if (bounds) {
+      const corner1 = latLng(bounds[1], bounds[0]),
+        corner2 = latLng(bounds[3], bounds[2]);
+      return latLngBounds(corner1, corner2);
```

Comment:
seems to be working but looks like it isn't? http://localhost:3000/catalog/princeton-fk4544658v-tilejson shows all black tiles, but https://map-tiles-staging.princeton.edu/2a91d82c541c426cb787cc62afe8f248/mosaicjson/tiles/WebMercatorQuad/5/18/12@1x returns those black tiles so I think this is working.